pyomo/solvers/tests/checks/test_cplex.py

```diff
@@ -54,5 +59,198 @@ def test_validate_file_name(self):
+        from pyomo.repn.plugins.cpxlp import ProblemWriter_cpxlp  # register the `WriterFactory`
+
+        self.mock_model = self.get_mock_model()
+        self.mock_cplex_shell = self.get_mock_cplex_shell(self.mock_model)
+        self.mock_cplex_shell._priorities_file_name = pyutilib.services.TempfileManager.create_tempfile(
+            suffix=".cplex.ord"
+        )
+
+    def tearDown(self):
+        pyutilib.services.TempfileManager.clear_tempfiles()
+
+    def get_mock_model(self):
+        model = ConcreteModel()
+        model.x = Var(within=Binary)
+        model.con = Constraint(expr=model.x >= 1)
+        model.obj = Objective(expr=model.x)
+        return model
+
+    def get_mock_cplex_shell(self, mock_model):
+        solver = MockCPLEX()
+        solver._problem_files, solver._problem_format, solver._smap_id = convert_problem(
+            (mock_model,),
+            ProblemFormat.cpxlp,
+            [ProblemFormat.cpxlp],
+            has_capability=lambda x: True,
+        )
+        return solver
+
+    def get_priorities_file_as_string(self, mock_cplex_shell):
+        with open(mock_cplex_shell._priorities_file_name, "r") as ord_file:
+            priorities_file = ord_file.read()
+        return priorities_file
+
+    def test_write_without_priority_suffix(self):
+        with self.assertRaises(ValueError):
+            CPLEXSHELL._write_priorities_file(self.mock_cplex_shell, self.mock_model)
+
+    def test_write_priority_to_priorities_file(self):
+        self.mock_model.priority = Suffix(direction=Suffix.EXPORT, datatype=Suffix.INT)
+        priority_val = 10
+        self.mock_model.priority.set_value(self.mock_model.x, priority_val)
+
+        CPLEXSHELL._write_priorities_file(self.mock_cplex_shell, self.mock_model)
+        priorities_file = self.get_priorities_file_as_string(self.mock_cplex_shell)
+
+        self.assertEqual(
+            priorities_file,
+            "* ENCODING=ISO-8859-1\nNAME             Priority Order\n  x1 %d\nENDATA\n"
+            % (priority_val,),
+        )
+
+    def test_write_priority_and_direction_to_priorities_file(self):
+        self.mock_model.priority = Suffix(direction=Suffix.EXPORT, datatype=Suffix.INT)
+        priority_val = 10
+        self.mock_model.priority.set_value(self.mock_model.x, priority_val)
+
+        self.mock_model.direction = Suffix(direction=Suffix.EXPORT, datatype=Suffix.INT)
+        direction_val = BranchDirection.down
+        self.mock_model.direction.set_value(self.mock_model.x, direction_val)
+
+        CPLEXSHELL._write_priorities_file(self.mock_cplex_shell, self.mock_model)
+        priorities_file = self.get_priorities_file_as_string(self.mock_cplex_shell)
+
+        self.assertEqual(
+            priorities_file,
+            "* ENCODING=ISO-8859-1\nNAME             Priority Order\n %s x1 %d\nENDATA\n"
+            % (ORDFileSchema._direction_to_str(direction_val), priority_val),
```

Comment:
I think I would prefer this string to be hard-coded (not generated with `%s` placeholders).  Doing so will ensure that this also tests that the `ORDFileSchema._direction_to_str` is correct.

Reply:
Sure. I've updated this and the below

Reply:
Below got updated, but I was thinking something closer to this here:
```python
self.assertEqual(
    priorities_file,
    "* ENCODING=ISO-8859-1\n"
    "NAME             Priority Order\n"
    " DN x1 10\n"
    "ENDATA\n")
```